### PR TITLE
docs: add a pull request template

### PR DESCRIPTION
**Pull request by HarshCasper**
**Please refer to the contribution guidelines in the README when submitting PRs.**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1 +1,32 @@
-**Please refer to the contribution guidelines in the README when submitting PRs.**
+# Description
+
+Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.
+
+Fixes # (issue)
```
Comment (whummer) [inline .github/PULL_REQUEST_TEMPLATE.md:5]: We should be aware that `Fixes #xxxx` automatically closes issues when merging PRs. Perhaps this is the desired behavior, so I'm in favor of it (if the rest of the team agrees). 👍 (Alternatively, we could use `Addresses #xxxx`)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1 +1,32 @@
...
+# How Has This Been Tested?
+
+Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+- [ ] Test A
+- [ ] Test B
+
+# Checklist:
+
+- [ ] My code follows the style guidelines of this project
```
Comment (whummer) [inline .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
- [ ] My code follows the style guidelines of this project (partially enforced by code linter in CI)
```

**Review by whummer — COMMENTED** [review]
Nice steps towards a more structured PR template @HarshCasper ! 👍 Only added a few minor comments - will let others chime in as well.. 🙌

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (whummer) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: (Maybe not necessary, but currently we don't have the style guidelines documented very well. E.g., the linter does currently not recommend using type hints, etc.)

**Comment by thrau** [comment]
closing this PR as stale - i think we should re-visit the template at some point though.

**Pull request closed without merging**